### PR TITLE
Fix parsing nested comments

### LINKED ISSUE #1026: Nested comments causes parser to timeout
Hi,

Large nested comments like below(produced by fuzzing);
```
{{#{{#{{#{{#{{#{{#{{#{{#{{#{{#{{#{{#{{#{{#{{##}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#}#
```
makes the parser get stuck at following lines.
https://github.com/djc/askama/blob/5955ba23c172ef6bbc0116b9ab0d6f068b86a688/askama_parser/src/node.rs#L1027-L1029

<br>

Steps to reproduce:
[comment-testcase.txt](https://github.com/djc/askama/files/15170493/comment-testcase.txt)

```rust
#[test]
fn test_fuzzed_comment_recursion() {
    const TEMPLATE: &str = include_str!("../../comment-testcase.txt");
    assert!(Ast::from_str(TEMPLATE, None, &Syntax::default()).is_err());
}
```
```sh
Executing task: cargo test --package askama_parser --lib -- tests::test_fuzzed_comment_recursion --exact --show-output 

   Compiling askama_parser v0.2.1 (/home/maxx/dev/security/oss-fuzz-projects/askama/askama_parser)
    Finished test [unoptimized + debuginfo] target(s) in 1.12s
     Running unittests src/lib.rs (target/debug/deps/askama_parser-c458ed7d8982bb6b)

running 1 test
```
<br> 

Will having a guard against deep nesting can be considered as a fix here ?

```diff
 use nom::multi::{fold_many0, many0, many1, separated_list0, separated_list1};
 use nom::sequence::{delimited, pair, preceded, terminated, tuple};

-use crate::{ErrorContext, ParseResult};
+use crate::{ErrorContext, Level, ParseResult};

 use super::{
     bool_lit, char_lit, filter, identifier, is_ws, keyword, num_lit, path_or_identifier, skip_till,
@@ -1025,6 +1025,9 @@ impl<'a> Comment<'a> {
                 let (end, tail) = take_until(s.syntax.comment_end)(i)?;
                 match take_until::<_, _, ErrorContext<'_>>(s.syntax.comment_start)(i) {
                     Ok((start, _)) if start.as_ptr() < end.as_ptr() => {
+                        if level >= Level::MAX_DEPTH {
+                            return Err(nom::Err::Failure(error_position!(i, ErrorKind::TooLarge)));
+                        }
                         level += 1;
                         i = &start[2..];
                     }
```

### PR DESCRIPTION
**Pull request by Kijewski**
This changes makes parsing nested comments run in linear time, and constant space.

Fixes #1026.

### CODE REVIEW

File: askama_parser/src/tests.rs [hunk 1]
```diff
@@ -918,3 +918,17 @@ fn fuzzed_unary_recursion() {
     const TEMPLATE: &str = include_str!("../tests/unary-recursion.txt");
     assert!(Ast::from_str(TEMPLATE, None, &Syntax::default()).is_err());
 }
+
+#[test]
```
Comment (Kijewski) [inline askama_parser/src/tests.rs:922]: Is there no annotation for `#[test]` to limit the runtime?

Comment (djc) [reply askama_parser/src/tests.rs:922]: No: https://doc.rust-lang.org/reference/attributes/testing.html.

File: askama_parser/src/node.rs [hunk 2]
```diff
@@ -1019,42 +1019,57 @@ pub struct Comment<'a> {
...
-                    }
-                    _ => return Ok((end, tail)),
+                let (_, (j, tag)) = skip_till(|i| tag(i, s))(i)?;
+                match tag {
+                    Tag::Open => match depth.checked_add(1) {
+                        Some(new_depth) => depth = new_depth,
+                        None => {
+                            return Err(nom::Err::Failure(ErrorContext {
+                                input: i,
+                                message: Some(Cow::Borrowed("comment nesting too big")),
```
Comment (djc) [inline askama_parser/src/node.rs:1045]: Nit: "comment nesting too big" -> "too deeply nested comments"?

**Review by djc — APPROVED** [review]
Thanks!

**Pull request merged**